src/cluster.cpp

```diff
@@ -128,13 +134,28 @@ void sync_start() {
 
   stop_signal = false;
 
+  nodesSearchedOthers = nodesSearchedSend = nodesSearchedRecv = 0;
+
   // Start listening to stop signal
   if (!is_root())
       MPI_Ibarrier(StopComm, &reqStop);
 }
 
 void sync_stop() {
 
+  // update the nodeCount, non-blocking collective,
+  // we don't quite now how many of those we issue on each rank, there might be pending a pending one on some ranks
+  int flag;
+  MPI_Test(&reqNodesSearched, &flag, MPI_STATUS_IGNORE);
+  if (flag)
+  {
+     nodesSearchedOthers = nodesSearchedRecv - nodesSearchedSend;
+     nodesSearchedSend = Threads.nodes_searched();
+     MPI_Iallreduce(&nodesSearchedSend, &nodesSearchedRecv, 1, MPI_UINT64_T,
```

Comment:
typo in comment: know
I suggest to use `StopComm` communicator here, since this is now a part of `sync_stop()`.